src/curl_internal.hpp

```diff
@@ -23,4 +23,25 @@ namespace powerloader::details
         CURLSetup& operator=(const CURLSetup&) = delete;
     };
 }
+
+namespace powerloader
+{
+    class CURLInterface
```

Comment:
I put this class here, but maybe having its own file is more appropriate so that we don't include `CURLSetup` where it is not needed.

Reply:
I think it's fine here (I think about it like a "module"), although not sure the impact of includes.
I'm fine with whatever you end up prefering.

Reply:
BTW, if the goal is just to have a name prefix, I think you can remove the class and replace it by a namespace.

Reply:
Hmm actually the `CURLInterface` functions have access to `CURLHandle` private curl stuff because they are friends. If we just use a namespace, is there a way to have access as well? (without including headers)

Reply:
Weird. Well the equivalent would be to make the functions as friend too. But then would check if it's really necessary to have these friends... or if maybe hiding a data structure in CURLHandle through a pointer which is known internally but not publicly wouldnt be a better solution.

Reply:
The thing is that we need to indeed hide a data structure like but we still need it to be accessible in many other places where it is used outside the `CURLHandle` (I guess that was the main reason why many of these curl functions were public in `CURLHandle`).

Reply:
Yes I meant that there is an alternative way to do it, I'll ping you in pms to discuss it.